gravwell/js-client
# feature(#18): Generate a docs website WITHOUT committing the static files

- Commits by @GeraldWicks
- Code review by @LucasPaganini, @Puszkarek, and @iagojsilva
- QA by @LucasPaganini 

---

Addresses: #18 

This PR has added an implementation to generate docs every time we run `npm run start:docs`, then go to `http://localhost:8080`. This will look at all folders inside `./src` and will generate docs inside `./docs`.

After this new PR. Nothing more is needed to generate the docs, the docs-generator loops through `./src` and creates the docs automatically.

I added comments in every file, but I'll also include a brief explanation of what each folder does here:
- assets: this folder has many files that will be copied to the docs folder, all these files are relevant to generate a nice homePage.
- files-manager: this folder helps to copy files, create files, delete files, edit files and also see all files inside a folder.
- folder-manager: this folder will help to create a folder and see all subfolders inside a folder.
- home-page: it has the .html file that will be copied to the docs folder
- pages: only has the Page interface
- readme: these readmes define a piece of the page that will show up, for example, on the functions page, the readme will represent this red line:
![Captura de tela de 2022-05-19 14-56-39](https://user-images.githubusercontent.com/69917459/169367359-a3284534-cad3-4d5c-b18a-58db147af29c.png)
- scripts: this folder will be responsible for all the scripts that will be added to the bottom of the .html pages, these scripts are important because they will change some default behaviors from typedoc lib (the lib used to generate these docs)
- settings: this folder will have some important files like blackListFolders, paths, and licenses.
- typedoc: on this folder, the typedoc lib will create the docs pages.
